pkg/pusher/pusher_test.go

```diff
@@ -35,60 +32,80 @@ import (
 // time to wait for received response from pushsync
 const spinTimeout = time.Second * 3
 
-var block = common.HexToHash("0x1").Bytes()
-var defaultMockValidStamp = func(ch swarm.Chunk, stamp []byte) (swarm.Chunk, error) {
-	return ch, nil
-}
-
-// createLocalstoreLock is used to prevent data race issues detected when multiple localstore.New functions
-// are being called in the tests.
-var createLocalstoreLock sync.Mutex
+var (
+	block                 = common.HexToHash("0x1").Bytes()
+	defaultMockValidStamp = func(ch swarm.Chunk, stamp []byte) (swarm.Chunk, error) {
+		return ch, nil
+	}
+	defaultRetryCount = 3
+)
 
-// Wrap the actual storer to intercept the modeSet that the pusher will call when a valid receipt is received
-type Store struct {
-	storage.Storer
-	internalStorer storage.Storer
-	modeSet        map[string]storage.ModeSet
-	modeSetMu      *sync.Mutex
+type mockStorer struct {
+	chunks         chan swarm.Chunk
+	reportedMu     sync.Mutex
+	reportedSynced []swarm.Chunk
+	reportedFailed []swarm.Chunk
+	reportedStored []swarm.Chunk
+}
 
-	closed              bool
-	setBeforeCloseCount int
-	setAfterCloseCount  int
+func (m *mockStorer) SubscribePush(ctx context.Context) (c <-chan swarm.Chunk, stop func()) {
+	return m.chunks, func() { close(m.chunks) }
 }
 
-// Override the Set function to capture the ModeSetSync
-func (s *Store) Set(ctx context.Context, mode storage.ModeSet, addrs ...swarm.Address) error {
-	s.modeSetMu.Lock()
-	defer s.modeSetMu.Unlock()
+func (m *mockStorer) Report(ctx context.Context, chunk swarm.Chunk, state storage.ChunkState) error {
+	m.reportedMu.Lock()
+	defer m.reportedMu.Unlock()
 
-	if s.closed {
-		s.setAfterCloseCount++
-	} else {
-		s.setBeforeCloseCount++
+	if state == storage.ChunkSynced {
+		m.reportedSynced = append(m.reportedSynced, chunk)
 	}
-
-	for _, addr := range addrs {
-		s.modeSet[addr.String()] = mode
+	if state == storage.ChunkCouldNotSync {
+		m.reportedFailed = append(m.reportedFailed, chunk)
+	}
+	if state == storage.ChunkStored {
+		m.reportedStored = append(m.reportedStored, chunk)
 	}
 	return nil
 }
 
-func (s *Store) Close() error {
-	s.modeSetMu.Lock()
-	defer s.modeSetMu.Unlock()
+func (m *mockStorer) isReported(chunk swarm.Chunk, state storage.ChunkState) bool {
+	m.reportedMu.Lock()
+	defer m.reportedMu.Unlock()
 
-	s.closed = true
-	return s.internalStorer.Close()
+	switch {
```

Comment:
You can put the `state` in the `switch` statement and in `case` check for particular `ChunkState` enum.